javascript/javascript1/week2/student-manager.js

```diff
@@ -1,20 +1,21 @@
 
 const class07Students = [];
 function addStudentToClass(studentName) {
-    if (studentName) {
-        if (class07Students.length < 6) {
-            if (class07Students.includes(studentName)) {
-                console.log(`Student ${studentName} is already in the class`)
-                }
-            else {
-                return class07Students.push(studentName)
-                } 
-        }else if (studentName === 'Queen'){
-            return class07Students.push('Queen')
-        }else {
-            console.log('Cannot add more students to class 07') 
-            }
+    if (!studentName) {
+        console.log(`"Please write down your name"`)
+        return
+    }else {
+    if (class07Students.length < 6 || studentName === 'Queen') {
+        if (class07Students.includes(studentName)) {
+            console.log(`Student ${studentName} is already in the class`)
         }
+        else {
+            return class07Students.push(studentName)
+        } 
+    }else {
+        console.log('Cannot add more students to class 07') 
+    }
```

Comment:
You can reduce nesting even further by taking every scenario that should NOT add a student and treat those first, one by one. I'll start:
```js
    if (!studentName) {
        console.log(`"Please write down your name"`)
        return
    }

    if (class07Students.length < 6 || studentName === 'Queen') {
        ...
```

Reply:
Thanks. I will do that.

Reply:
I try to reduce nesting but I m not sure about my solution.